examples/0_basic/6_create_vp.rs

```diff
@@ -0,0 +1,198 @@
+  // Step 5: Holder creates and signs a verifiable presentation from the issued credential.
+  // ===========================================================================
+
+  // Deserialize the credential.
+  let credential: Credential = Credential::from_json(credential_json.as_str())?;
+
+  // Create an unsigned Presentation from the previously issued Verifiable Credential.
+  let mut presentation: Presentation = PresentationBuilder::default()
+    .holder(Url::parse(alice_document.id().as_ref())?)
+    .credential(credential)
+    .build()?;
+
+  // Sign the verifiable presentation using the holder's verification method
+  // and include the requested challenge and expiry timestamp.
+  alice_document.sign_data(
+    &mut presentation,
+    key_pair_alice.private(),
+    "#key-1",
+    ProofOptions::new().challenge(challenge.to_string()).expires(expires),
+  )?;
+
+  // ===========================================================================
+  // Step 6: Holder sends a verifiable presentation to the verifier.
+  // ===========================================================================
+
+  // Convert the Verifiable Presentation to JSON to send it to the verifier.
+  let presentation_json: String = presentation.to_json()?;
+
+  // ===========================================================================
+  // Step 7: Verifier receives the Verifiable Presentation and verifies it.
+  // ===========================================================================
+
+  // Deserialize the presentation from the holder:
+  let presentation: Presentation = Presentation::from_json(&presentation_json)?;
+
+  // The verifier wants the following requirements to be satisfied:
+  // - Signature verification (including checking the requested challenge to mitigate replay attacks)
+  // - Presentation validation must fail if credentials expiring within the next 10 hours are encountered
+  // - The presentation holder must always be the subject, regardless of the presence of the nonTransferable property
+  // - The issuance date must not be in the future.
+
+  let presentation_verifier_options: VerifierOptions = VerifierOptions::new()
+    .challenge(challenge.to_owned())
+    .allow_expired(false);
+
+  // Do not allow credentials that expire within the next 10 hours.
+  let credential_validation_options: CredentialValidationOptions = CredentialValidationOptions::default()
+    .earliest_expiry_date(Timestamp::now_utc().checked_add(Duration::hours(10)).unwrap());
+
+  let presentation_validation_options = PresentationValidationOptions::default()
+    .presentation_verifier_options(presentation_verifier_options.clone())
+    .shared_validation_options(credential_validation_options)
+    .subject_holder_relationship(SubjectHolderRelationship::AlwaysSubject);
+
+  PresentationValidator::validate(
+    &presentation,
+    &alice_document,
+    &[issuer_document],
+    &presentation_validation_options,
+    FailFast::FirstError,
+  )?;
```

Comment:
I think it might be better to use the `Resolver::verify_presentation` API here and demonstrate that it will resolve any DID documents it needs that are not passed in. Perhaps the example can assume that the verifier already has Alice's document, but not the issuers?